staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1319,15 +1319,26 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 				return []watch.Event{{Type: watch.Added, Object: createdPodsAfterWatch[0]}}
 			},
 		},
+		{
+			name:                               "sendInitialEvents=false, RV=17",
+			sendInitialEvents:                  &falseVal,
+			resourceVersion:                    "17",
+			initialPods:                        func(ns string) []*example.Pod { return []*example.Pod{makePod(ns, "18"), makePod(ns, "19")} },
+			podsAfterEstablishingWatch:         func(ns string) []*example.Pod { return []*example.Pod{makePod(ns, "20")} },
+			expectedInitialEventsInRandomOrder: addEventsFromCreatedPods,
+			expectedEventsAfterEstablishingWatch: func(createdPodsAfterWatch []*example.Pod) []watch.Event {
+				return []watch.Event{{Type: watch.Added, Object: createdPodsAfterWatch[0]}}
+			},
+		},
 		{
 			name:                               "legacy, RV=0",
 			resourceVersion:                    "0",
-			initialPods:                        func(ns string) []*example.Pod { return []*example.Pod{makePod(ns, "18"), makePod(ns, "19")} },
+			initialPods:                        func(ns string) []*example.Pod { return []*example.Pod{makePod(ns, "21"), makePod(ns, "22")} },
```

Comment:
I already asked for that before - let's not enumerate pods globally per test - given we're using namespaces, let's use "1, 2, ..." independently for each case [ideally in a separate PR]